src/type-info.c

```diff
@@ -82,13 +82,17 @@ static enum vctrs_type vec_base_typeof(SEXP x, bool proxied) {
   case CPLXSXP: return vctrs_type_complex;
   case STRSXP: return vctrs_type_character;
   case RAWSXP: return vctrs_type_raw;
+  case S4SXP:
+    // S4 objects are only vectors if they are proxied
+    if (proxied) return vctrs_type_list;
```

Comment:
👍

Reply:
Actually can you move this to a separate PR? The rando class is not a S4SXP.

Reply:
Done. We'll need to revert 15cbb55a44 in that separate PR.